pydicom/pixel_data_handlers/pillow_handler.py

```diff
@@ -80,6 +80,33 @@ def should_change_PhotometricInterpretation_to_RGB(dicom_dataset):
     return False
 
 
+def _decompress_single_frame(data, transfer_syntax, photometric_interpretation):
+    fio = io.BytesIO(data)
+    try:
+        image = Image.open(fio)
+        # This hack ensures that RGB color images, which were not
+        # color transformed (i.e. not transformed into YCbCr color space)
+        # upon JPEG compression are decompressed correctly.
+        # Since Pillow assumes that images were transformed into YCbCr color
+        # space prior to compression, setting the value of "mode" to YCbCr
+        # signals Pillow to not apply any color transformation upon
+        # decompression.
+        if (transfer_syntax in PillowJPEGTransferSyntaxes and
```

Comment:
This could just be `if transfer_syntax.is_compressed ...` since `get_pixeldata()` already checks that its a valid transfer syntax.